corehq/ex-submodules/casexml/apps/phone/tests/test_async_restore.py

```diff
@@ -146,16 +147,17 @@ def test_completed_task_deletes_cache(self):
         restore_config.timing_context.start()
         restore_config.timing_context("wait_for_task_to_start").start()
         get_async_restore_payload.delay(restore_config)
-        self.assertTrue(restore_config.timing_context.is_finished())
         self.assertIsNone(async_restore_task_id_cache.get_value())
 
     def test_completed_task_creates_sync_log(self):
         restore_config = self._restore_config(is_async=True)
         restore_config.timing_context.start()
         restore_config.timing_context("wait_for_task_to_start").start()
         get_async_restore_payload.delay(restore_config)
-        self.assertTrue(restore_config.timing_context.is_finished())
-        self.assertIsNotNone(restore_config.restore_state.current_sync_log)
+        sync_logs = SyncLogSQL.objects.filter(domain='dummy-project')
+        self.assertEqual(len(sync_logs), 1)
```

Comment:
Is it true that the number of sync logs before this test is run should be zero? What do you think of adding a check for that as well?

Reply:
https://github.com/dimagi/commcare-hq/pull/31584/commits/b1e66c1a4e15b4603dc22bcefb1c5948d2cbb833